pkg/game/elements/placed_tile.go

```diff
@@ -0,0 +1,112 @@
+package elements
+
+import (
+	"fmt"
+
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tiles"
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tiles/side"
+	"github.com/YetAnotherSpieskowcy/Carcassonne-Engine/pkg/tilesets"
+)
+
+type Position struct {
+	// int8 would be fine for base game (72 tiles) but let's be a bit more generous
+	x int16
+	y int16
+}
+
+func NewPosition(x int16, y int16) Position {
+	return Position{x, y}
+}
+
+func (pos Position) X() int16 {
+	return pos.x
+}
+
+func (pos Position) Y() int16 {
+	return pos.y
+}
+
+func (pos Position) MarshalText() ([]byte, error) {
+	return fmt.Appendf([]byte{}, "%v,%v", pos.x, pos.y), nil
+}
+
+func (pos *Position) UnmarshalText(text []byte) error {
+	_, err := fmt.Sscanf(string(text), "%v,%v", &pos.x, &pos.y)
+	return err
+}
+
+// https://wikicarpedia.com/car/Game_Figures
+type MeepleType uint8
+
+const (
+	NormalMeeple MeepleType = iota
+
+	MeepleTypeCount int = iota
+)
+
+// represents a legal position (and rotation) of a tile on the board
+type TilePlacement struct {
+	tiles.Tile
+	Pos Position
+}
+
+// note: this is implemented to avoid inheriting Tile's implementation
+// but is just a simple passthrough and its output may not be a legal placement.
+func (placement TilePlacement) Rotate(rotations uint) TilePlacement {
+	return TilePlacement{
+		Tile: placement.Tile.Rotate(rotations),
+		Pos:  placement.Pos,
+	}
+}
+
+// represents a legal position of a meeple on the tile
+type MeeplePlacement struct {
+	Side side.Side
+	Type MeepleType
+}
+
+// note: this is implemented to aid with LegalMove.Rotate() implementation
+// but is just a simple passthrough and its output may not be a legal placement.
+func (placement MeeplePlacement) Rotate(rotations uint) MeeplePlacement {
+	return MeeplePlacement{
+		Side: placement.Side.Rotate(rotations),
+		Type: placement.Type,
+	}
+}
+
+// represents a legal move (tile placement and meeple placement) on the board
+type LegalMove struct {
+	TilePlacement
+	// LegalMove always has a `Meeple`. Whether it is actually placed
+	// is determined by `MeeplePlacement.Side` which will be `None`, if it isn't.
+	Meeple MeeplePlacement
+}
+
+// note: this is implemented to avoid inheriting the implementation
+// but is just a simple passthrough and its output may not be a legal move.
+func (move LegalMove) Rotate(rotations uint) LegalMove {
```

Comment:
If calling this method may cause "LegalMove" to no longer be legal then we shouldn't allow this api call to be possible in the first place and just panic out to  break inheritance line.

Reply:
Done.